crates/ruma-events/src/room/message/url_preview.rs

```diff
@@ -4,9 +4,13 @@ use crate::room::{EncryptedFile, OwnedMxcUri, UInt};
 
 /// The Source of the PreviewImage.
 #[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
```

Comment:
This is actually fine as an exhaustive enum (like [`MediaSource`](https://docs.ruma.dev/ruma/events/room/enum.MediaSource.html))

```suggestion
#[allow(clippy::exhaustive_enums)]
```